File: two-sum/two-sum.go
import "fmt"

func twoSum(nums []int, target int) []int {
    var seen_variables  = map[int] int {
        
    }
    for i:=0;i<len(nums);i++  {
        difference := target - nums[i]
        value, ok := seen_variables[difference]
        if ok  {
            fmt.Println("value: ", value , "seem" , seen_variables , "Difference", difference)
            solution := [] int {seen_variables[difference],i};
            fmt.Println(solution)
            return solution
        }
        seen_variables[nums[i]] = i 
    }
    var no_value [] int;
    return no_value
}

File: two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        target_differences = {}
        for i,j in enumerate(nums):
            difference = target - j
            if difference in target_differences:
                return [target_differences[difference],i]
            
            target_differences[j] = i 